pkg/kubectl/cmd/edit.go

```diff
@@ -354,7 +350,9 @@ func getPrinter(cmd *cobra.Command) (*editPrinterOptions, error) {
 	}
 }
 
-func getMapperAndResult(f cmdutil.Factory, args []string, options *resource.FilenameOptions, editMode EditMode) (meta.RESTMapper, *resource.Mapper, *resource.Result, string, error) {
+type resultGetter func([]byte) *resource.Result
```

Comment:
Why not return a function that returns a resource.InfoListVisitor here instead of making the caller cast it in from the result?